File: AnySqlWebAdmin/Code/Math/cVector_3d.cs

namespace Vectors
{

    public class cVector_3d
    {
        public bool bCurrentlyValid = true;
        public double x = 0;
        public double y = 0;
        public double z = 0;


        //Constructor
        public cVector_3d(double nXparam = 0, double nYparam = 0, double nZparam = 0)
        {
            this.bCurrentlyValid = true;
            this.x = nXparam;
            this.y = nYparam;
            this.z = nZparam;
        } // End Constructor



        // cVector_3d.MakeVector(endx, endy, endz [, startx, starty, starz]);
        public static cVector_3d MakeVector(double nX1param, double nY1param, double nZ1param, double nX0param = 0, double nY0param = 0, double nZ0param = 0)
        {
            cVector_3d vecReturnValue = new cVector_3d(nX1param - nY0param, nY1param - nY0param, nZ1param - nZ0param);
            return vecReturnValue;
        } // End function MakeVector


        // cVector_3d.VectorSubtract(vec2, vec1);
        public static cVector_3d VectorSubtract(cVector_3d b, cVector_3d a)
        {
            cVector_3d vecReturnValue = new cVector_3d(b.x - a.x, b.y - a.y, b.z - a.z);
            return vecReturnValue;
        } // End function VectorSubtract


        // cVector_3d.VectorAdd(vec1, vec2);
        public static cVector_3d VectorAdd(cVector_3d a, cVector_3d b)
        {
            cVector_3d vecReturnValue = new cVector_3d(a.x + b.x, a.y + b.y, a.z + b.z);
            return vecReturnValue;
        } // End function VectorAdd



        // cVector_3d.CrossP(vec1, vec2);
        public static cVector_3d CrossP(cVector_3d a, cVector_3d b)
        {
            //A × B = [(ay*bz-az*by),(az*bx-ax*bz),(ax*by-ay*bx)]
            cVector_3d vecReturnValue = new cVector_3d((a.y * b.z - a.z * b.y), (a.z * b.x - a.x * b.z), (a.x * b.y - a.y * b.x));
            return vecReturnValue;
        } // End function CrossP



        // cVector_3d.DotP(vec1, vec2);
        public static double DotP(cVector_3d a, cVector_3d b)
        {
            //A * B = ax*bx+ay*by+az*bz
            double nReturnValue = a.x * b.x + a.y * b.y + a.z * b.z;
            return nReturnValue;
        } // End function DotP



        // cVector_3d.VectorLength(vec);
        public static double VectorLength(cVector_3d vec)
        {
            double nReturnValue = System.Math.Pow(vec.x, 2) + System.Math.Pow(vec.y, 2) + System.Math.Pow(vec.z, 2);
            nReturnValue = System.Math.Sqrt(nReturnValue);
            return nReturnValue;
        } // End function VectorLength



        // cVector_3d.VectorNorm(vec);
        public static double VectorNorm(cVector_3d vec)
        {
            return VectorLength(vec);
        } // End function VectorNorm



        // cVector_3d.CrossP(vec1, vec2);
        public static cVector_3d UnitVector(cVector_3d vec)
        {
            double len = VectorLength(vec);
            cVector_3d vecReturnValue = new cVector_3d(vec.x / len, vec.y / len, vec.z / len);
            return vecReturnValue;
        } // End function UnitVector



        // cVector_3d.Angle_Rad(vec1, vec2);
        public static double Angle_Rad(cVector_3d a, cVector_3d b)
        {
            double nReturnValue =
            System.Math.Acos(
                        (
                            (a.x * b.x + a.y * b.y + a.z * b.z) /
                                (System.Math.Sqrt(System.Math.Pow(b.x, 2) + System.Math.Pow(b.y, 2) + System.Math.Pow(b.z, 2))
                                * System.Math.Sqrt(System.Math.Pow(a.x, 2) + System.Math.Pow(a.y, 2) + System.Math.Pow(a.z, 2))
                            )
                        )

                    );
            return nReturnValue;
        }  // End function Angle_Rad


        // cVector_3d.Angle_Degrees(vec1, vec2);
        public static double Angle_Degrees(cVector_3d a, cVector_3d b)
        {
            double nReturnValue =
            System.Math.Acos(
                        (
                            (a.x * b.x + a.y * b.y + a.z * b.z) /
                                (System.Math.Sqrt(System.Math.Pow(b.x, 2) + System.Math.Pow(b.y, 2) + System.Math.Pow(b.z, 2))
                                * System.Math.Sqrt(System.Math.Pow(a.x, 2) + System.Math.Pow(a.y, 2) + System.Math.Pow(a.z, 2))
                            )
                        )

                    );
            return nReturnValue * 180 / System.Math.PI;
        }  // End function Angle_Degrees



    } // End cPoint 

} // End Package

// http://mathworld.wolfram.com/Point-LineDistance3-Dimensional.html
// http://mathworld.wolfram.com/Point-LineDistance2-Dimensional.html

File: AnySqlWebAdminOld/wwwroot/ts/Math/cVector_3d.js
var Vectors;
(function (Vectors) {
    var cVector_3d = (function () {
        function cVector_3d(x0, y0, z0) {
            if (x0 === void 0) { x0 = 0; }
            if (y0 === void 0) { y0 = 0; }
            if (z0 === void 0) { z0 = 0; }
            this.bCurrentlyValid = true;
            this.x = 0;
            this.y = 0;
            this.z = 0;
            this.bCurrentlyValid = true;
            this.x = x0;
            this.y = y0;
            this.z = z0;
        }
        cVector_3d.MakeVector = function (x1, y1, z1, x0, y0, z0) {
            if (x0 === void 0) { x0 = 0; }
            if (y0 === void 0) { y0 = 0; }
            if (z0 === void 0) { z0 = 0; }
            var vecReturnValue = new cVector_3d(x1 - x0, y1 - y0, z1 - z0);
            return vecReturnValue;
        };
        cVector_3d.VectorSubtract = function (b, a) {
            var vecReturnValue = new cVector_3d(b.x - a.x, b.y - a.y, b.z - a.z);
            return vecReturnValue;
        };
        cVector_3d.VectorAdd = function (a, b) {
            var vecReturnValue = new cVector_3d(a.x + b.x, a.y + b.y, a.z + b.z);
            return vecReturnValue;
        };
        cVector_3d.CrossP = function (a, b) {
            var vecReturnValue = new cVector_3d((a.y * b.z - a.z * b.y), (a.z * b.x - a.x * b.z), (a.x * b.y - a.y * b.x));
            return vecReturnValue;
        };
        cVector_3d.DotP = function (a, b) {
            var nReturnValue = a.x * b.x + a.y * b.y + a.z * b.z;
            return nReturnValue;
        };
        cVector_3d.VectorLength = function (vec) {
            var nReturnValue = Math.pow(vec.x, 2) + Math.pow(vec.y, 2) + Math.pow(vec.z, 2);
            nReturnValue = Math.sqrt(nReturnValue);
            return nReturnValue;
        };
        cVector_3d.VectorNorm = function (vec) {
            return cVector_3d.VectorLength(vec);
        };
        cVector_3d.UnitVector = function (vec) {
            var len = cVector_3d.VectorLength(vec);
            var vecReturnValue = new cVector_3d(vec.x / len, vec.y / len, vec.z / len);
            return vecReturnValue;
        };
        cVector_3d.Angle_Rad = function (a, b) {
            var nReturnValue = Math.acos(((a.x * b.x + a.y * b.y + a.z * b.z) /
                (Math.sqrt(Math.pow(b.x, 2) + Math.pow(b.y, 2) + Math.pow(b.z, 2))
                    * Math.sqrt(Math.pow(a.x, 2) + Math.pow(a.y, 2) + Math.pow(a.z, 2)))));
            return nReturnValue;
        };
        cVector_3d.Angle_Degrees = function (a, b) {
            var nReturnValue = Math.acos(((a.x * b.x + a.y * b.y + a.z * b.z) /
                (Math.sqrt(Math.pow(b.x, 2) + Math.pow(b.y, 2) + Math.pow(b.z, 2))
                    * Math.sqrt(Math.pow(a.x, 2) + Math.pow(a.y, 2) + Math.pow(a.z, 2)))));
            return nReturnValue * 180 / Math.PI;
        };
        return cVector_3d;
    }());
    Vectors.cVector_3d = cVector_3d;
})(Vectors || (Vectors = {}));
